src/test/scala/za/co/absa/commons/io/TempFileSpec.scala

```diff
@@ -56,7 +56,9 @@ class TempFileSpec extends AnyFlatSpec with Matchers {
 
   it should "return valid URI" in {
     val tempFile = TempFile().deleteOnExit()
-    val expectedURIString = s"file:/${tempFile.path.toAbsolutePath.toString.replace("\\", "/")}"
+    val expectedURIString = s"file:/${tempFile.path.toAbsolutePath.toString
+      .replace("\\", "/")
+      .replace("//", "/")}"
```

Comment:
To my taste this look rather complicated especially in a place where a hardcoded literal (expected value) would be ideal.

I would like to suggest the following change. Since we are testing the structure of the actual returned value (path), we need to put more control on function dependencies to rule out all possible non-deterministic behavior and predict the output with certainty. Specifically, the given function has two such side variables:
1. the location of the system temp directory
2. the auto-generated unique filename

The location of the temp dir we could reliably get from the `java.io.tmpdir` system property and use it to construct the expected value. But to abstract from the auto-generated filenames we could introduce a factory function would work as a _humble object_ - it would simply delegate to the native `Files.createTempFile` but would be easily mockable in tests.

See my suggestion PR to this branch - #122

Reply:
Proposal merged. Added required logic because of windows. See c60ddf8e2e3f8b2420904fdc098d8623c0b541ea.